Sources/MaliciousSiteProtection/API/APIClient.swift

```diff
@@ -18,129 +18,119 @@
-            logDebug("🔸 Invalid matches URL: \(hashPrefix)")
-            return []
-        }
-        return await fetch(url: url, responseType: MatchResponse.self)?.matches ?? []
+    public func load<Request: APIRequestProtocol>(_ requestConfig: Request) async throws -> Request.ResponseType {
+        let requestType = requestConfig.requestType
+        let headers = environment.headers(for: requestType)
+        let url = environment.url(for: requestType)
+
+        let apiRequest = APIRequestV2(url: url, method: .get, headers: headers)
+        let response = try await service.fetch(request: apiRequest)
+        let result: Request.ResponseType = try response.decodeBody()
+
+        return result
     }
-}
 
-// MARK: Private Methods
-extension APIClient {
+}
 
-    private func logDebug(_ message: String) {
-        Logger.api.debug("\(message)")
+// MARK: - Convenience
+extension APIClientProtocol {
+    public func filtersChangeSet(for threatKind: ThreatKind, revision: Int) async throws -> APIClient.Response.FiltersChangeSet {
+        let result = try await load(.filterSet(threatKind: threatKind, revision: revision))
+        return result
     }
 
-    private func createURL(for path: Constants.APIPath, revision: Int? = nil, queryItems: [URLQueryItem]? = nil) -> URL? {
-        // Start with the base URL and append the path component
-        var urlComponents = URLComponents(url: endpointURL.appendingPathComponent(path.rawValue), resolvingAgainstBaseURL: true)
-        var items = queryItems ?? []
-        if let revision = revision, revision > 0 {
-            items.append(URLQueryItem(name: "revision", value: String(revision)))
-        }
-        urlComponents?.queryItems = items.isEmpty ? nil : items
-        return urlComponents?.url
+    public func hashPrefixesChangeSet(for threatKind: ThreatKind, revision: Int) async throws -> APIClient.Response.HashPrefixesChangeSet {
+        let result = try await load(.hashPrefixes(threatKind: threatKind, revision: revision))
+        return result
     }
 
-    private func fetch<T: Decodable>(url: URL, responseType: T.Type) async -> T? {
-        var request = URLRequest(url: url)
-        request.httpMethod = "GET"
-        request.allHTTPHeaderFields = headers
-
-        do {
-            let (data, _) = try await session.data(for: request)
-            if let response = try? JSONDecoder().decode(responseType, from: data) {
-                return response
-            } else {
-                logDebug("🔸 Failed to decode response for \(String(describing: responseType)): \(data)")
-            }
-        } catch {
-            logDebug("🔴 Failed to load \(String(describing: responseType)) data: \(error)")
-        }
-        return nil
+    public func matches(forHashPrefix hashPrefix: String) async throws -> APIClient.Response.Matches {
+        let result = try await load(.matches(hashPrefix: hashPrefix))
```

Comment:
Can we pass a shorter timeout to the /matches endpoint? This could block navigation so we should favour navigation loading if the backend is degraded. On Android we're looking at a maximum 1 second timeout for this request. The other requests can be heavier but they are background tasks anyway so not as risky.

Reply:
Added parametrized Timeout